src/Refinery/ProofState.hs

```diff
@@ -301,7 +317,7 @@ subgoals fs (Effect m) = Effect (fmap (subgoals fs) m)
 subgoals fs (Stateful s) = Stateful (fmap (subgoals fs) . s)
 subgoals fs (Alt p1 p2) = Alt (subgoals fs p1) (subgoals fs p2)
 subgoals fs (Interleave p1 p2) = Interleave (subgoals fs p1) (subgoals fs p2)
-subgoals fs (Commit p1 p2) = Commit (subgoals fs p1) (subgoals fs p2)
+-- subgoals fs (Commit p1 p2 k) = Commit (subgoals fs p1) (subgoals fs p2)
```

Comment:
Not entirely sure what's going on here, but i think it needs the `blarg` treatment